### ISSUE TITLE
Installing while the application is opened make a computer restart

### ISSUE THREAD

**Issue by marcobeninca71** [oleg-shilo/wixsharp#470]
Dear Oleg
I don't know if you are the rigth person to answer but I'm trying to understand a very strange behaviour of my installation package. The behaviour is as follow:
- I have my product already installed on the PC
- I run my installation package to upgrade it to a new version while the application is opened
- During the installation process the package close the application and upgrade the software correctly
- But in the case that the application has pending modifications to the opened document it asks the user to save or discard. In the case the user choose discard then after having installed the new version the installation makes the PC to restart.

Any idea?
Thanks in advance
Marco

**Comment by oleg-shilo** [comment]
Hi Marco,

I haven't seen this very behavior before. Though I have observer behavior that is conceptually similar. For example when some files (e.g service) is locked the msi may ask user to reboot. That's expected. But I have also seen it rebooting without the confirmation.

But I cannot really give you a proper explanation for that. Sorry mate.

Cheers,
Oleg

**Comment by marcobeninca71** [comment]
Hi Oleg
Thanks for your qick reply. I investigated a little bit deeper and I found that:

- If the user answer immediatelly to the dialog asking to save the modifications the PC doesn't restart

- If the user wait for a while (I don't really know how much) then after the end of the installation the PC restart

It looks like the installation system has a certain timeout after which the PC has to be restarted to be sure that everything is installed correctly.

Is there a way to set this timeout or to disable it? Is there a way to always prevent a restart? In my case a restart is never necessary

Marco
